disableWebSecurity -> allowDisplayingInsecureContent

jnugh opened the pull request:
Allow mixed content does not use disableWebSecurity anymore.

We don't use a sledgehammer to crack a nut when this gets merged. Disabling security features is usually a bad idea - in this case we do not need to disable all security features to reach the goal.
As of electron 1.4.5 we can disable only certain security features as we need to.

This also fixes the youtube preview issue which was related to the disabled CORS security feature.

---

- [x] Complete [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
- [x] Execute `npm run prettify` to format codes
- [x] Update `CHANGELOG.md` and/or `doc/*.md` if it's necessary.
- [x] Write about environment which you tested
  - Ubuntu 16.10 Unity

yuya-oc commented:
Tested on macOS 10.12. The feature works well!

On the https server (pre-release.mattermost.com), they appear:
- "http"(insecure) images
- youtube preview

yuya-oc commented on src/browser/index.jsx:
```diff
@@ -343,7 +343,7 @@ var MattermostView = React.createClass({
     // This option disables the same-origin policy and allows js/css/plugins not only content like images.
```
Please update.

yuya-oc commented on src/browser/index.jsx:
```diff
@@ -343,7 +343,7 @@ var MattermostView = React.createClass({
     // This option disables the same-origin policy and allows js/css/plugins not only content like images.
     if (config.disablewebsecurity === true) {
       // webview.setAttribute('disablewebsecurity', false) disables websecurity. (electron's bug?)
```
Should be removed.

yuya-oc commented on CHANGELOG.md:
```diff
@@ -23,6 +23,8 @@ Release date: TBD
 ### Bug Fixes
  - Fixed wrong cursor for "Edit" and "Remove" in Setting page
  - Fixed an issue where "Zoom in/out" does not properly work
+ - "Allow mixed content" does not disable web-security generally
```
Is there any other exact description? For example, ""Allow mixed content" did not disable insecure css/javascript". Alternatively, ""Allow mixed content" now allows only insecure images" in Improvements section. Any thoughts?

yuya-oc commented:
You can find the artifacts to test here: https://circleci.com/gh/mattermost/desktop/768#artifacts

jasonblais commented:
Tested on 64-bit Windows installer and works great. Big thanks @jnugh! 

Test cases:
 - Mixed content enabled: `http`, `https` images and YouTube videos render
 - Mixed content disabled: `http` images don't render, YouTube videos and `https` images do

--

I assume the build failure is due to the test script Yuya mentions above.

![image](https://cloud.githubusercontent.com/assets/13119842/20032622/8455ba76-a364-11e6-931e-f0dacfeba232.png)

jasonblais commented:
One note: Anything else we should test with the Electron dependency bump from 1.4.2 --> 1.4.5? 

I noticed that the zoom in/out issue [@yuya-oc described here](https://github.com/mattermost/desktop/pull/240#issuecomment-254197719) no longer reproduces. If I click on the tab bar or the chat area, zooming in and out works properly. 

The zoom rendering issues only reproduce when restarting the app as described in https://github.com/mattermost/desktop/issues/334.

yuya-oc commented:
@jasonblais As far as reading changelog, I can't find significant change for our app.

I tested the app on Windows 10 again. I think that the issue still reproduces when clicking the tab bar. Would you check again?

jasonblais commented:
I tried again on Windows 10, and I was able to reproduce it after clicking on the tab bar twice.

![image](https://cloud.githubusercontent.com/assets/13119842/20157971/c39c67ea-a6a5-11e6-944f-2a02715f5c56.png)
*After clicking on tab bar once*

![image](https://cloud.githubusercontent.com/assets/13119842/20158015/ef77e0a6-a6a5-11e6-9b3f-8d543b34ae7e.png)
*After clicking on tab bar twice*

jnugh commented:
Just rebased and pushed multiple changes:
* Updated the tests - no failures anymore on my machine
* Changed some comments inside the code, as requested in the review, I didn't change the code itself
* Updated Changelog to be more specific about the security features disabled previously

jnugh commented:
https://circleci.com/gh/mattermost/desktop/780#artifacts :)

yuya-oc commented:
In `package.json`, Electron is not 1.4.5 probably due to rebasing. Please check.

jnugh commented:
You are right, seems like I confused the prebuilt package version I removed during merge with the resulting one :disappointed: 
Sorry for that - https://circleci.com/gh/mattermost/desktop/781#artifacts should have a working version in a few seconds.

jasonblais commented:
Sorry @jnugh, it seems like the build is failing, can you help take a look?

jnugh commented:
Seems like CircleCI (or the the Docker Repo) is having issues right now.
Err https://apt.dockerproject.org ubuntu-trusty/main amd64 Packages
  gnutls_handshake() failed: Handshake failed

I restarted twice by pushing a new ref - will try again later.

yuya-oc commented:
I restarted CircleCI, and the test is passing.

https://circleci.com/gh/mattermost/desktop/784#artifacts

jasonblais commented:
Test cases:
 - Mixed content enabled: `http`, `https` images and YouTube videos render
 - Mixed content disabled: `http` images don't render, YouTube videos and `https` images do

All passed on 64-bit Windows installer :+1: 

Thanks @jnugh :)

yuya-oc commented:
Tested on macOS, the test cases passed!

yuya-oc commented on CHANGELOG.md:
```diff
@@ -24,6 +24,8 @@ Release date: TBD
 ### Bug Fixes
  - Fixed wrong cursor for "Edit" and "Remove" in Setting page
  - Fixed an issue where "Zoom in/out" does not properly work
+ - The "Allow mixed content" setting used to disable multiple security features to allow http content
```
@jnugh Are there any updates for here?

jnugh replied:
It means: allowRunningInsecureContent is disabled and CORS is enabled now.

jasonblais replied:
Cool, wondering how we might be able to translate it to a non-technical end user or admin. Would this be accurate?

```
The "Allow mixed content" setting that enables http content no longer disables existing features such as YouTube previews
```

jnugh replied:
Which is what the next line says ;) I will simply remove this note and leave

>  - YouTube preview works, even if mixed content is allowed

jasonblais commented:
Thanks @jnugh! :)

yuya-oc commented:
@jnugh Thanks many times!
